src/components/Resource/Forms/Training/TrainingSubmit.tsx

```diff
@@ -496,12 +500,45 @@ export const TrainingFormSubmit = ({ model }: { model: NonNullable<TrainingModel
                 await router.replace(userTrainingDashboardURL);
               },
               onError: (error) => {
-                // TODO [bw] set status back to pending
-                setAwaitInvalidate(false);
-                showErrorNotification({
-                  error: new Error(error.message),
-                  title: 'Failed to submit.',
-                });
+                // set the status back to pending
+                upsertVersionMutation.mutate(
+                  {
+                    ...basicVersionData,
+                    baseModel: baseModelConvert,
+                    trainingStatus: TrainingStatus.Pending,
+                  },
+                  {
+                    async onSuccess(response, request) {
+                      queryUtils.training.getModelBasic.setData({ id: model.id }, (old) => {
+                        if (!old) return old;
+
+                        const versionToUpdate = old.modelVersions.find(
+                          (mv) => mv.id === thisModelVersion.id
+                        );
+                        if (!versionToUpdate) return old;
+
+                        versionToUpdate.trainingStatus = request.trainingStatus!;
+
+                        return {
+                          ...old,
+                          modelVersions: [
+                            versionToUpdate,
+                            ...old.modelVersions.filter((mv) => mv.id !== thisModelVersion.id),
+                          ],
+                        };
+                      });
+                      // TODO [bw] don't invalidate, just update
+                      await queryUtils.model.getMyTrainingModels.invalidate();
+                    },
+                    onSettled() {
+                      setAwaitInvalidate(false);
+                      showErrorNotification({
+                        error: new Error(error.message),
+                        title: 'Failed to submit.',
+                      });
+                    },
```

Comment:
I feel that this can be placed outside of this mutation so the user gets visual feedback as soon as the first request fails. I might be missing some context tho 😛

Reply:
`doTraining` has an error notification already, this one is for the version update. ill just change the language to make it a bit more clear.